notebooks/Learning_Rate_Scheduler.ipynb

```diff
@@ -0,0 +1,3360 @@
+      "    967        \u001b[36m0.4469\u001b[0m  0.0001  0.0014\n",
+      "    968        \u001b[36m0.4469\u001b[0m  0.0001  0.0019\n",
+      "    969        \u001b[36m0.4469\u001b[0m  0.0001  0.0018\n",
+      "    970        \u001b[36m0.4469\u001b[0m  0.0001  0.0014\n",
+      "    971        \u001b[36m0.4469\u001b[0m  0.0001  0.0014\n",
+      "    972        \u001b[36m0.4469\u001b[0m  0.0001  0.0017\n",
+      "    973        \u001b[36m0.4469\u001b[0m  0.0001  0.0019\n",
+      "    974        \u001b[36m0.4469\u001b[0m  0.0001  0.0013\n",
+      "    975        \u001b[36m0.4469\u001b[0m  0.0001  0.0014\n",
+      "    976        \u001b[36m0.4469\u001b[0m  0.0001  0.0014\n",
+      "    977        \u001b[36m0.4469\u001b[0m  0.0001  0.0017\n",
+      "    978        \u001b[36m0.4469\u001b[0m  0.0001  0.0017\n",
+      "    979        \u001b[36m0.4469\u001b[0m  0.0001  0.0017\n",
+      "    980        \u001b[36m0.4469\u001b[0m  0.0001  0.0017\n",
+      "    981        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    982        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    983        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    984        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    985        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    986        \u001b[36m0.4469\u001b[0m  0.0000  0.0016\n",
+      "    987        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    988        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    989        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    990        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    991        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    992        \u001b[36m0.4469\u001b[0m  0.0000  0.0016\n",
+      "    993        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    994        \u001b[36m0.4469\u001b[0m  0.0000  0.0018\n",
+      "    995        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    996        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    997        \u001b[36m0.4469\u001b[0m  0.0000  0.0017\n",
+      "    998        0.4469  0.0000  0.0017\n",
+      "    999        0.4469  0.0000  0.0019\n",
+      "   1000        0.4469  0.0000  0.0019\n",
+      "CosineAnnealingLR Score: 0.4500\n",
+      "\n",
+      "Final Results Summary:\n",
+      "\n",
+      "Scheduler: StepLR\n",
+      "Score: 0.4500\n",
+      "Simulated Learning Rates: [0.05 0.05 0.05 0.05 0.05]...\n",
+      "\n",
+      "Scheduler: ReduceLROnPlateau\n",
+      "Score: 0.4000\n",
+      "Simulated Learning Rates: [0.05 0.05 0.05 0.05 0.05]...\n",
+      "\n",
+      "Scheduler: CosineAnnealingLR\n",
+      "Score: 0.4500\n",
+      "Simulated Learning Rates: [0.05       0.04999988 0.04999951 0.04999889 0.04999803]...\n"
+     ]
+    }
+   ],
+   "source": [
+    "def simulate_scheduler(scheduler_class, optimizer_params, steps, initial_lr):\n",
```

Comment:
So I checked the skorch code again and I think we can greatly simplify this. Normally, we record the learning rates in the `history`, which means we can just do training and once we finish, we can check `net.history[:, 'event_lr']` to find the learning rates over time. There is no extra code needed to record the learning rates.

The only caveat to this is that we would not record the lr for `ReduceLROnPlateau`. I created a PR to enable this: #1075. So I'd say let's wait for this PR to be merged, then change the notebook to take the learning rates from the history.

Reply:
Please feel free to ping me when #1075, gets merged and I will make the required changes.

Reply:
The PR is now merged. This means after you finish training a net, you can call `net.history[:, 'event_lr']` to retrieve the learning rates. Please give it a try. Remember to merge the latest master branch of skorch into your branch.

Reply:
Updated as discussed.